src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1553,5 +1553,7 @@ INamedTypeSymbolInternal INamedTypeSymbolInternal.EnumUnderlyingType
                 return this.EnumUnderlyingType;
             }
         }
+
+        internal override bool IsRecord => false;
```

Comment:
>internal override bool IsRecord => false; [](start = 8, length = 41)

Adding this implementation to a base type is error prone. Leaf types deriving from NamedTypeSymbol should implement this API instead.

Reply:
Updated - thanks so much for the feedback so far.